setup.py

```diff
@@ -52,26 +54,48 @@
 )
 
 
-py_version = sys.version[:3]
+py_version = sys.version_info
+is_pypy = "PyPy" in sys.version
 
 SCRIPT1 = 'nose2'
-SCRIPT2 = 'nose2-%s' % (py_version,)
+SCRIPT2 = 'nose2-%s.%s' % (py_version.major, py_version.minor)
+
 
+def _get_specific_requirements():
+    directory = dirname(__file__)
+    version_specific_requirements_path = 'requirements-{}.{}.txt'.format(py_version.major, py_version.minor)
+    if is_pypy:
+        pypy_name = "pypy"
+        if py_version.major == 3:
+            pypy_name = 'pypy3'
+        version_specific_requirements_path = 'requirements-{}.txt'.format(pypy_name)
+    if not os.path.exists(os.path.join(directory, version_specific_requirements_path)):
+
+        if py_version.major == 3:
+            version_specific_requirements_path = 'requirements-3.5.txt'
+        else:
+            version_specific_requirements_path = 'requirements-2.7.txt'
+    with open(os.path.join(directory, version_specific_requirements_path)) as requirements_file:
```

Comment:
Maybe use `pip.req`?
http://jelly.codes/articles/python-pip-module/

Reply:
Otherwise, this should at least be a function to avoid duplicate code.

Reply:
Yes, I will deduplicate. Thanks for pip.req, I didn't know it :)

Reply:
in fact `pip.req` seems not to be a good idea : https://stackoverflow.com/questions/14399534/reference-requirements-txt-for-the-install-requires-kwarg-in-setuptools-setup-py : 

>  You really don't want to do this. Speaking as a pip maintainer pip does not support being called as an API like this at all. In fact pip 1.6 (next version at this time) moves this function. – Donald Stufft Mar 26 '14 at 0:59

Reply:
You are correct.
but :-( would be nice if they supported this